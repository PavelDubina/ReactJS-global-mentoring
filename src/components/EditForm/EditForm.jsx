import React from 'react'
import { useDispatch } from 'react-redux'
import { Formik } from 'formik'
import * as yup from 'yup'
import PropTypes from 'prop-types'
import { Button } from '../Button/Button'
import { FormSelect } from '../FormSelect/FormSelect'
import { Modal } from '../Modal/Modal'
import { Input } from '../Input/Input'
import editMovie from '../../redux/actions/editMovie'
import styles from './EditForm.scss'

export const EditForm = ({ onClose, ...props }) => {
  const dispatch = useDispatch()
  const validationSchema = yup.object().shape({
    title: yup.string().required('Required'),
    poster_path: yup.string().required('Required').url('Enter correct url, like as http://exmple.com'),
    release_date: yup.date().max(new Date(), 'Release date must be at earler than today').required('Required'),
    genres: yup.array().min(1, 'Select at least one genre to be proceed'),
    overview: yup.string().required('Required'),
    runtime: yup
      .number()
      .required('Required')
      .positive('Runtime must be a positive number')
      .integer('Runtime must be an integer'),
  })
  const initialValues = {
    title: props.title,
    id: props.id,
    release_date: props.release_date,
    overview: props.overview,
    runtime: props.runtime || '',
    genres: props.genres.map((genre) => genre.toLowerCase()),
    poster_path: props.poster_path,
  }
  return (
    <Modal title="EDIT MOVIE" onClose={onClose}>
      <Formik
        initialValues={initialValues}
        onSubmit={(values) => {
          dispatch(editMovie({ ...props, ...values }))
          onClose()
        }}
        validationSchema={validationSchema}
      >
        {({ values, errors, handleChange, handleSubmit, handleReset }) => {
          const { id, title, release_date, overview, runtime, genres, poster_path } = values
          return (
            <>
              <form className={styles.form}>
                <Input title="MOVIE ID" name="id" type="number" value={id} disabled />
                <Input title="TITLE" name="title" type="text" value={title} placeholder="Title here" />
                <br />
                <Input title="RELEASE DATE" name="release_date" type="date" value={release_date} />
                <br />
                <Input
                  title="POSTER PATH"
                  name="poster_path"
                  type="text"
                  value={poster_path}
                  placeholder="Poster URL here"
                />
                <br />
                <FormSelect name="genres" value={genres.join(', ')} onChange={handleChange} />
                {errors.genres && <p className={styles.error}>{errors.genres}</p>}
                <br />
                <Input title="OVERVIEW" name="overview" type="text" value={overview} placeholder="Overview here" />
                <br />
                <Input title="RUNTIME" name="runtime" type="number" value={runtime} placeholder="Runtime here" />
              </form>
              <div className={styles.btns}>
                <Button onClick={handleReset} styleType="reset">
                  RESET
                </Button>
                <Button type="submit" onClick={handleSubmit} styleType="confirm">
                  SAVE
                </Button>
              </div>
            </>
          )
        }}
      </Formik>
    </Modal>
  )
}

EditForm.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  runtime: PropTypes.number,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  poster_path: PropTypes.string,
  vote_average: PropTypes.string,
  vote_count: PropTypes.string,
  onClose: PropTypes.func.isRequired,
}
