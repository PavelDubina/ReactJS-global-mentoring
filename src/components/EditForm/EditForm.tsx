import React from 'react'
import { useDispatch } from 'react-redux'
import { Formik } from 'formik'
import * as yup from 'yup'
import { useTranslation } from 'next-i18next'
import { Button, EStyleTypeBtn, EType } from '../Button/Button'
import { FormSelect } from '../FormSelect/FormSelect'
import { Modal } from '../Modal/Modal'
import { Input } from '../Input/Input'
import editMovie from '../../redux/actions/editMovie'
import styles from '../../../styles/EditForm.module.scss'

type EditFormProps = {
  id: number
  title: string
  release_date: string
  overview: string
  runtime: number | string
  genres: string[]
  poster_path: string
  vote_average: number
  vote_count: number
  onClose: () => void
}

// PATTERN:{Destructuring props}
// PATTERN:{JSX spread attributes}
export const EditForm: React.FC<EditFormProps> = ({ onClose, ...props }) => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const validationSchema = yup.object().shape({
    title: yup.string().required(t('editModal.validation.title')),
    poster_path: yup
      .string()
      .required(t('editModal.validation.poster.required'))
      .url(t('editModal.validation.poster.url')),
    release_date: yup
      .date()
      .max(new Date(), t('editModal.validation.date.max'))
      .required(t('editModal.validation.date.required')),
    genres: yup.array().min(1, t('editModal.validation.genre')),
    overview: yup.string().required(t('editModal.validation.overview')),
    runtime: yup
      .number()
      .required(t('editModal.validation.runtime.required'))
      .positive(t('editModal.validation.runtime.positive'))
      .integer(t('editModal.validation.runtime.integer')),
  })
  const initialValues = {
    title: props.title,
    id: props.id,
    release_date: props.release_date,
    overview: props.overview,
    runtime: props.runtime || '',
    genres: props.genres.map((genre) => genre.toLowerCase()),
    poster_path: props.poster_path,
  }
  return (
    <Modal title={t('editModal.title')} onClose={onClose}>
      <Formik
        initialValues={initialValues}
        onSubmit={(values) => {
          dispatch(editMovie({ ...props, ...values }))
          onClose()
        }}
        validationSchema={validationSchema}
      >
        {({ values, errors, handleChange, handleSubmit, handleReset }) => {
          const { id, title, release_date, overview, runtime, genres, poster_path } = values
          return (
            <>
              <form className={styles.form}>
                <Input title={t('editModal.input.id')} name="id" type="number" value={id} disabled />
                <Input
                  title={t('editModal.input.title')}
                  name="title"
                  type="text"
                  value={title}
                  placeholder={t('editModal.placeholders.title')}
                />
                <br />
                <Input title={t('editModal.input.date')} name="release_date" type="date" value={release_date} />
                <br />
                <Input
                  title={t('editModal.input.poster')}
                  name="poster_path"
                  type="text"
                  value={poster_path}
                  placeholder={t('editModal.placeholders.poster')}
                />
                <br />
                <FormSelect name="genres" value={genres.join(', ')} onChange={handleChange} />
                {errors.genres && <p className={styles.error}>{errors.genres}</p>}
                <br />
                <Input
                  title={t('editModal.input.overview')}
                  name="overview"
                  type="text"
                  value={overview}
                  placeholder={t('editModal.placeholders.overview')}
                />
                <br />
                <Input
                  title={t('editModal.input.runtime')}
                  name="runtime"
                  type="number"
                  value={runtime}
                  placeholder={t('editModal.placeholders.runtime')}
                />
              </form>
              <div className={styles.btns}>
                <Button onClick={handleReset} styleType={EStyleTypeBtn.reset}>
                  {t('editModal.reset')}
                </Button>
                <Button type={EType.submit} onClick={handleSubmit} styleType={EStyleTypeBtn.confirm}>
                  {t('editModal.submit')}
                </Button>
              </div>
            </>
          )
        }}
      </Formik>
    </Modal>
  )
}
