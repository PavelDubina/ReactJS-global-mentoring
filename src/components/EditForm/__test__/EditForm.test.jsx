import React from 'react'
import configureStore from 'redux-mock-store'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { Provider } from 'react-redux'
import movies from '../../../MockedData/mockMovies.json'
import { EditForm } from '../EditForm'

const mockStore = configureStore([])

describe('EditForm', () => {
  const store = mockStore({
    moviesData: movies,
  })
  ReactDOM.createPortal = jest.fn((element) => element)
  const props = {
    title: 'test title',
    id: 1,
    release_date: '12/12/2020',
    overview: 'test overview',
    runtime: 100,
    genres: ['comedy'],
    poster_path: 'test path',
  }
  const getComponent = ({ onClose = jest.fn() }) =>
    render(
      <Provider store={store}>
        <EditForm onClose={onClose} {...props} />
      </Provider>,
    )
  it('Should renders correctly', () => {
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('Should renders correctly without runtime value', () => {
    const props = {
      title: 'test title',
      id: 1,
      release_date: '12/12/2020',
      overview: 'test overview',
      genres: ['comedy'],
      poster_path: 'test path',
    }
    const getComponent = ({ onClose = jest.fn() }) =>
      render(
        <Provider store={store}>
          <EditForm onClose={onClose} {...props} />
        </Provider>,
      )
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('Should show validation on blur and empty field', async () => {
    const { getByTestId } = getComponent({})
    const input = getByTestId('title')
    fireEvent.change(input, { target: { value: '' } })
    fireEvent.blur(input)
    await waitFor(() => {
      expect(getByTestId('error')).not.toBe(null)
      expect(getByTestId('error')).toHaveTextContent('Required')
    })
  })
})
