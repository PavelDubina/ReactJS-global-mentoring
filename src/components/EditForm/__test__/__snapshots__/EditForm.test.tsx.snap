// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditForm Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="modal_backdrop"
  >
    <div
      class="modal_window"
    >
      <div
        class="close_btn"
      >
        ×
      </div>
      <h1
        class="title"
      >
        editModal.title
      </h1>
      <form
        class="form"
      >
        <label
          class="label"
        >
          editModal.input.id
          <input
            autocomplete="off"
            class="input"
            data-testid="id"
            disabled=""
            name="id"
            type="number"
            value="1"
          />
        </label>
        <label
          class="label"
        >
          editModal.input.title
          <input
            autocomplete="off"
            class="input"
            data-testid="title"
            name="title"
            placeholder="editModal.placeholders.title"
            type="text"
            value="test title"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.date
          <input
            autocomplete="off"
            class="input"
            data-testid="release_date"
            name="release_date"
            type="date"
            value="12/12/2020"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.poster
          <input
            autocomplete="off"
            class="input"
            data-testid="poster_path"
            name="poster_path"
            placeholder="editModal.placeholders.poster"
            type="text"
            value="test path"
          />
        </label>
        <br />
        <span>
          addModal.input.genre
        </span>
        <div
          class="container"
        >
          <div
            class="select"
          >
            <p
              class="value"
            >
              addModal.genres.comedy
            </p>
            <div
              class="arrow"
            >
              ▼
            </div>
          </div>
          <div
            class="dropdown"
            data-testid="dropdown"
          >
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="action"
                name="genres"
                type="checkbox"
                value="action"
              />
              <label
                class="lable"
                for="action"
              >
                <span>
                  addModal.genres.action
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="adventure"
                name="genres"
                type="checkbox"
                value="adventure"
              />
              <label
                class="lable"
                for="adventure"
              >
                <span>
                  addModal.genres.adventure
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="animation"
                name="genres"
                type="checkbox"
                value="animation"
              />
              <label
                class="lable"
                for="animation"
              >
                <span>
                  addModal.genres.animation
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                checked=""
                class="checkbox"
                id="comedy"
                name="genres"
                type="checkbox"
                value="comedy"
              />
              <label
                class="lable"
                for="comedy"
              >
                <span>
                  addModal.genres.comedy
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="crime"
                name="genres"
                type="checkbox"
                value="crime"
              />
              <label
                class="lable"
                for="crime"
              >
                <span>
                  addModal.genres.crime
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="documentary"
                name="genres"
                type="checkbox"
                value="documentary"
              />
              <label
                class="lable"
                for="documentary"
              >
                <span>
                  addModal.genres.documentary
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="drama"
                name="genres"
                type="checkbox"
                value="drama"
              />
              <label
                class="lable"
                for="drama"
              >
                <span>
                  addModal.genres.drama
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="family"
                name="genres"
                type="checkbox"
                value="family"
              />
              <label
                class="lable"
                for="family"
              >
                <span>
                  addModal.genres.family
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="fantasy"
                name="genres"
                type="checkbox"
                value="fantasy"
              />
              <label
                class="lable"
                for="fantasy"
              >
                <span>
                  addModal.genres.fantasy
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="history"
                name="genres"
                type="checkbox"
                value="history"
              />
              <label
                class="lable"
                for="history"
              >
                <span>
                  addModal.genres.history
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="horror"
                name="genres"
                type="checkbox"
                value="horror"
              />
              <label
                class="lable"
                for="horror"
              >
                <span>
                  addModal.genres.horror
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="music"
                name="genres"
                type="checkbox"
                value="music"
              />
              <label
                class="lable"
                for="music"
              >
                <span>
                  addModal.genres.music
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="mystery"
                name="genres"
                type="checkbox"
                value="mystery"
              />
              <label
                class="lable"
                for="mystery"
              >
                <span>
                  addModal.genres.mystery
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="romance"
                name="genres"
                type="checkbox"
                value="romance"
              />
              <label
                class="lable"
                for="romance"
              >
                <span>
                  addModal.genres.romance
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="science fiction"
                name="genres"
                type="checkbox"
                value="science fiction"
              />
              <label
                class="lable"
                for="science fiction"
              >
                <span>
                  addModal.genres.science fiction
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="thriller"
                name="genres"
                type="checkbox"
                value="thriller"
              />
              <label
                class="lable"
                for="thriller"
              >
                <span>
                  addModal.genres.thriller
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="war"
                name="genres"
                type="checkbox"
                value="war"
              />
              <label
                class="lable"
                for="war"
              >
                <span>
                  addModal.genres.war
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="western"
                name="genres"
                type="checkbox"
                value="western"
              />
              <label
                class="lable"
                for="western"
              >
                <span>
                  addModal.genres.western
                </span>
              </label>
            </div>
          </div>
        </div>
        <br />
        <label
          class="label"
        >
          editModal.input.overview
          <input
            autocomplete="off"
            class="input"
            data-testid="overview"
            name="overview"
            placeholder="editModal.placeholders.overview"
            type="text"
            value="test overview"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.runtime
          <input
            autocomplete="off"
            class="input"
            data-testid="runtime"
            name="runtime"
            placeholder="editModal.placeholders.runtime"
            type="number"
            value="100"
          />
        </label>
      </form>
      <div
        class="btns"
      >
        <button
          class="reset"
          type="button"
        >
          editModal.reset
        </button>
        <button
          class="confirm"
          type="submit"
        >
          editModal.submit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EditForm Should renders correctly without runtime value 1`] = `
<DocumentFragment>
  <div
    class="modal_backdrop"
  >
    <div
      class="modal_window"
    >
      <div
        class="close_btn"
      >
        ×
      </div>
      <h1
        class="title"
      >
        editModal.title
      </h1>
      <form
        class="form"
      >
        <label
          class="label"
        >
          editModal.input.id
          <input
            autocomplete="off"
            class="input"
            data-testid="id"
            disabled=""
            name="id"
            type="number"
            value="1"
          />
        </label>
        <label
          class="label"
        >
          editModal.input.title
          <input
            autocomplete="off"
            class="input"
            data-testid="title"
            name="title"
            placeholder="editModal.placeholders.title"
            type="text"
            value="test title"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.date
          <input
            autocomplete="off"
            class="input"
            data-testid="release_date"
            name="release_date"
            type="date"
            value="12/12/2020"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.poster
          <input
            autocomplete="off"
            class="input"
            data-testid="poster_path"
            name="poster_path"
            placeholder="editModal.placeholders.poster"
            type="text"
            value="test path"
          />
        </label>
        <br />
        <span>
          addModal.input.genre
        </span>
        <div
          class="container"
        >
          <div
            class="select"
          >
            <p
              class="value"
            >
              addModal.genres.comedy
            </p>
            <div
              class="arrow"
            >
              ▼
            </div>
          </div>
          <div
            class="dropdown"
            data-testid="dropdown"
          >
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="action"
                name="genres"
                type="checkbox"
                value="action"
              />
              <label
                class="lable"
                for="action"
              >
                <span>
                  addModal.genres.action
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="adventure"
                name="genres"
                type="checkbox"
                value="adventure"
              />
              <label
                class="lable"
                for="adventure"
              >
                <span>
                  addModal.genres.adventure
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="animation"
                name="genres"
                type="checkbox"
                value="animation"
              />
              <label
                class="lable"
                for="animation"
              >
                <span>
                  addModal.genres.animation
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                checked=""
                class="checkbox"
                id="comedy"
                name="genres"
                type="checkbox"
                value="comedy"
              />
              <label
                class="lable"
                for="comedy"
              >
                <span>
                  addModal.genres.comedy
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="crime"
                name="genres"
                type="checkbox"
                value="crime"
              />
              <label
                class="lable"
                for="crime"
              >
                <span>
                  addModal.genres.crime
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="documentary"
                name="genres"
                type="checkbox"
                value="documentary"
              />
              <label
                class="lable"
                for="documentary"
              >
                <span>
                  addModal.genres.documentary
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="drama"
                name="genres"
                type="checkbox"
                value="drama"
              />
              <label
                class="lable"
                for="drama"
              >
                <span>
                  addModal.genres.drama
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="family"
                name="genres"
                type="checkbox"
                value="family"
              />
              <label
                class="lable"
                for="family"
              >
                <span>
                  addModal.genres.family
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="fantasy"
                name="genres"
                type="checkbox"
                value="fantasy"
              />
              <label
                class="lable"
                for="fantasy"
              >
                <span>
                  addModal.genres.fantasy
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="history"
                name="genres"
                type="checkbox"
                value="history"
              />
              <label
                class="lable"
                for="history"
              >
                <span>
                  addModal.genres.history
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="horror"
                name="genres"
                type="checkbox"
                value="horror"
              />
              <label
                class="lable"
                for="horror"
              >
                <span>
                  addModal.genres.horror
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="music"
                name="genres"
                type="checkbox"
                value="music"
              />
              <label
                class="lable"
                for="music"
              >
                <span>
                  addModal.genres.music
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="mystery"
                name="genres"
                type="checkbox"
                value="mystery"
              />
              <label
                class="lable"
                for="mystery"
              >
                <span>
                  addModal.genres.mystery
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="romance"
                name="genres"
                type="checkbox"
                value="romance"
              />
              <label
                class="lable"
                for="romance"
              >
                <span>
                  addModal.genres.romance
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="science fiction"
                name="genres"
                type="checkbox"
                value="science fiction"
              />
              <label
                class="lable"
                for="science fiction"
              >
                <span>
                  addModal.genres.science fiction
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="thriller"
                name="genres"
                type="checkbox"
                value="thriller"
              />
              <label
                class="lable"
                for="thriller"
              >
                <span>
                  addModal.genres.thriller
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="war"
                name="genres"
                type="checkbox"
                value="war"
              />
              <label
                class="lable"
                for="war"
              >
                <span>
                  addModal.genres.war
                </span>
              </label>
            </div>
            <div
              class="input"
            >
              <input
                class="checkbox"
                id="western"
                name="genres"
                type="checkbox"
                value="western"
              />
              <label
                class="lable"
                for="western"
              >
                <span>
                  addModal.genres.western
                </span>
              </label>
            </div>
          </div>
        </div>
        <br />
        <label
          class="label"
        >
          editModal.input.overview
          <input
            autocomplete="off"
            class="input"
            data-testid="overview"
            name="overview"
            placeholder="editModal.placeholders.overview"
            type="text"
            value="test overview"
          />
        </label>
        <br />
        <label
          class="label"
        >
          editModal.input.runtime
          <input
            autocomplete="off"
            class="input"
            data-testid="runtime"
            name="runtime"
            placeholder="editModal.placeholders.runtime"
            type="number"
            value="120"
          />
        </label>
      </form>
      <div
        class="btns"
      >
        <button
          class="reset"
          type="button"
        >
          editModal.reset
        </button>
        <button
          class="confirm"
          type="submit"
        >
          editModal.submit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
